ocaml/xapi/xapi_vdi.ml

```diff
@@ -189,15 +199,7 @@ let check_operation_error ~__context ?(sr_records=[]) ?(pbd_records=[]) ?(vbd_re
           if List.mem record.Db_actions.vDI_type [ `ha_statefile; `redo_log ]
           then Some (Api_errors.operation_not_allowed, ["VDI containing HA statefile or redo log cannot be copied (check the VDI's allowed operations)."])
           else None
-        | `clone ->
-          if not Smint.(has_capability Vdi_clone sm_features)
-          then Some (Api_errors.sr_operation_not_supported, [Ref.string_of sr])
-          else None
-        | `mirror ->
-          if not Smint.(has_capability Vdi_mirror sm_features)
-          then Some (Api_errors.sr_operation_not_supported, [Ref.string_of sr])
-          else None
-        | _ -> None
+        | `mirror | `clone | `generate_config | `scan | `force_unlock | `blocked | `update -> None
```

Comment:
According to the Real World OCaml book, catch-all cases are especially bad with polymorphic variants: https://realworldocaml.org/v1/en/html/variants.html#polymorphic-variants :smile: